File: Models/Budget.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MilaAPI.Models
{
	public class Budget
	{
		[Key]
		public int Id { get; set; }

		[Required]
		public int UserId { get; set; }
		public User User { get; set; }  // Relation with User

		[Required]
		public int CategoryId { get; set; }
		public Category Category { get; set; }  // Relation with Category

		[Column(TypeName = "decimal(18,2)")]
		public decimal Amount { get; set; }  // Budget amount
		public DateTime StartDate { get; set; }
		public DateTime EndDate { get; set; }
		public bool IsActive { get; set; }
		
		[Column(TypeName = "decimal(18,2)")]
		public decimal CurrentAmount { get; set; }

		[Column(TypeName = "decimal(18,2)")]
		public decimal Limit { get; set; }
		public string Name { get; set; }
	}
}

File: front-end/milaapp/lib/models/budget.dart
class Budget {
  final String id;
  final double totalAmount;
  final double currentAmount;
  final DateTime startDate;
  final DateTime endDate;
  final String userId;

  Budget({
    required this.id,
    required this.totalAmount,
    required this.currentAmount,
    required this.startDate,
    required this.endDate,
    required this.userId,
  });

  factory Budget.fromJson(Map<String, dynamic> json) {
    return Budget(
      id: json['id'],
      totalAmount: json['totalAmount'],
      currentAmount: json['currentAmount'],
      startDate: DateTime.parse(json['startDate']),
      endDate: DateTime.parse(json['endDate']),
      userId: json['userId'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'totalAmount': totalAmount,
      'currentAmount': currentAmount,
      'startDate': startDate.toIso8601String(),
      'endDate': endDate.toIso8601String(),
      'userId': userId,
    };
  }
}
